### PR TITLE
Feat/more complex functions

### PR DESCRIPTION
**Pull request by jtlap**
some new complex function and fixing corner cases
also modyfing exp (real) and ulpdist

### CODE REVIEW

File: include/eve/module/complex/regular/detail/math.hpp [hunk 1]
```diff
@@ -126,6 +167,37 @@ namespace eve::detail
...
+  //===-------------------------------------------------------------------------------------------
+  template<typename Z>
+  EVE_FORCEINLINE auto complex_unary_dispatch( eve::tag::sin_
+                                             , Z const& z) noexcept
+  {
+    auto j = i(as(z));
+    return -(j*sinh(j*z));
+  }
+
+//   template<typename Z>
```
Comment (jfalcou) [inline include/eve/module/complex/regular/detail/math.hpp:182]: remove don't comment

File: test/unit/module/complex/measures.hpp [hunk 2]
```diff
@@ -17,14 +17,25 @@ namespace tts
     auto diff = l-r;
     auto [rl, il] = l;
     auto [rr, ir] = r;
-
+    auto udr = eve::ulpdist(il, ir);
+    auto udi = eve::ulpdist(rl, rr);
+//     std::cout << "        udi " << udi << std::endl;
```
Comment (jfalcou) [inline test/unit/module/complex/measures.hpp:22]: remove don't comment

File: test/unit/module/complex/measures.hpp [hunk 3]
```diff
@@ -17,14 +17,25 @@ namespace tts
...
-                          , eve::ulpdist(rl, rr)
+                          , udr
                           , eve::if_else( eve::almost(eve::is_imag)(diff)
-                                        , eve::ulpdist(il, ir)
-                                        , eve::ulpdist(l, r)
+                                        , udi
+                                        , eve::max(udi, udr)
                                         )
                           );
+//     std::cout << " alisr    " << eve::almost(eve::is_real)(diff)<< std::endl;
```
Comment (jfalcou) [inline test/unit/module/complex/measures.hpp:33]: remove don't comment

File: test/unit/module/complex/sin.cpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+EVE_TEST_TYPES( "Check corner cases of sin", eve::test::scalar::ieee_reals)
+  <typename T>(eve::as<T>)
+{
+  using e_t = eve::element_type_t<T>;
+  using c_t = eve::complex<e_t>;
+  using eve::as;
+  const int N = 11;
+  auto zer = eve::zero(as<e_t>());
+  auto inf = eve::inf(as<e_t>());
+//  auto minf = eve::minf(as<e_t>());
```
Comment (jfalcou) [inline test/unit/module/complex/sin.cpp:70]: remove don't comment

File: test/unit/module/complex/sqrt.cpp [hunk 5]
```diff
@@ -63,77 +63,74 @@ EVE_TEST( "Check behavior of sqrt on wide"
...
+      c_t(nan,nan)  //13
+    };
+
+  using eve::conj;
+  using eve::sqrt;
+  for(int i=0; i < N; ++i)
   {
-    ce_t z = cv(std::sqrt(c_t(a0.get(i), a1.get(i))));
-    b.set(i, z);
+    std::cout << "i " << i << " -> " << inputs[i] << std::endl;
```
Comment (jfalcou) [inline test/unit/module/complex/sqrt.cpp:120]: superfluous I/O

File: test/unit/module/complex/sqrt.cpp [hunk 6]
```diff
@@ -151,6 +148,6 @@ EVE_TEST_TYPES( "Check return types of eve::sqrt", eve::test::scalar::ieee_reals
   }
    TTS_ULP_EQUAL(eve::sqrt(c_t(-4)), c_t(0, 2), 0);
    TTS_ULP_EQUAL(eve::sqrt(T(-4)), eve::nan(as<T>()), 0);
-//  TTS_ULP_EQUAL(eve::cmplx(eve::sqrt)(T(-4)), c_t(0, 2), 0);
+//TTS_ULP_EQUAL(eve::cmplx(eve::sqrt)(T(-4)), c_t(0, 2), 0);
```
Comment (jfalcou) [inline test/unit/module/complex/sqrt.cpp:151]: remove don't comment

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/complex/regular/detail/math.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/complex/regular/detail/math.hpp:182]: done

File: test/unit/module/complex/measures.hpp (hunk 3 above)
Comment (jtlap) [reply test/unit/module/complex/measures.hpp:33]: done

File: test/unit/module/complex/sin.cpp (hunk 4 above)
Comment (jtlap) [reply test/unit/module/complex/sin.cpp:70]: done

Comment (jtlap) [reply test/unit/module/complex/sin.cpp:70]: done

File: test/unit/module/complex/sqrt.cpp (hunk 5 above)
Comment (jtlap) [reply test/unit/module/complex/sqrt.cpp:120]: done

**Review by jfalcou — APPROVED** [review]
